### PR TITLE
AQ-43 cypress

### PR DESCRIPTION
**Pull request by ToJestKrzysio**
# Description
Added cypress for headless testing. Starting GUI from docker container is most likely gonna cause more problems than it's gonna solve, due to X server configuration.

# Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The related memme has been added
- [x] The related memme has been added

image

image

### CODE REVIEW

File: quiz/docker-compose.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 services:
-  quiz_web:
+  web:
```
Comment (ToJestKrzysio) [inline quiz/docker-compose.yaml:2]: removed quiz prefix, 
Compose by default prefixes all services with the folder name which caused all of the services to be created as 
`quiz-quiz_<name>-1`

Comment (martynawitkowska) [reply quiz/docker-compose.yaml:2]: I find it better, it is more readable in the terminal. Services names were added after @pawelkonior review. It was said then that services name would look better with project names in docker-compose.yaml files

Comment (ToJestKrzysio) [reply quiz/docker-compose.yaml:2]: `quiz-quiz_web-1`

Comment (ToJestKrzysio) [reply quiz/docker-compose.yaml:2]: `quiz-web-1`

Comment (martynawitkowska) [reply quiz/docker-compose.yaml:2]: `quiz-web-1` is much better. No argue there 😄

File: quiz/envs/postgres.default.env [hunk 2]
```diff
@@ -1,7 +1,7 @@
 POSTGRES_USER=
 POSTGRES_PASSWORD=
 POSTGRES_DB=
-POSTGRES_HOST=
+POSTGRES_HOST=postgres
 POSTGRES_PORT=
```
Comment (martynawitkowska) [inline quiz/envs/postgres.default.env:5]: I think we can also add a default port for postgres 5432

File: quiz/web/cypress/e2e/spec.cy.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+describe('template spec', () => {
+  it('passes', () => {
+    cy.visit('https://example.cypress.io')
+  })
+})
```
Comment (martynawitkowska) [inline quiz/web/cypress/e2e/spec.cy.ts:5]: new line

File: quiz/web/cypress/support/commands.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+// declare global {
+//   namespace Cypress {
+//     interface Chainable {
+//       login(email: string, password: string): Chainable<void>
+//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
+//     }
+//   }
+// }
```
Comment (martynawitkowska) [inline quiz/web/cypress/support/commands.ts:37]: new line

File: quiz/readme.md [hunk 5]
```diff
@@ -13,3 +13,35 @@
 2. Copy `envs/postgres.default.env` to `envs/postgres.env` and fill environmental variables.
 3. Copy `envs/backend.default.env` to `envs/backend.env` and fill environmental variables.
 4. Start services `docker compose -f docker-compose.prod.yaml up --build`.
+
+## Testing
+
```
Comment (martynawitkowska) [inline quiz/readme.md:18]: Quick question why not running tests through for e. q. `docker compose exec <service name> pytest`?

File: quiz/envs/postgres.default.env (hunk 2 above)
Comment (ToJestKrzysio) [reply quiz/envs/postgres.default.env]: added

File: quiz/web/cypress/e2e/spec.cy.ts (hunk 3 above)
Comment (ToJestKrzysio) [reply quiz/web/cypress/e2e/spec.cy.ts]: done

File: quiz/web/cypress/support/commands.ts (hunk 4 above)
Comment (ToJestKrzysio) [reply quiz/web/cypress/support/commands.ts]: done

File: quiz/readme.md (hunk 5 above)
Comment (ToJestKrzysio) [reply quiz/readme.md:18]: yeah also valid approach

**Review by pawelkonior — APPROVED** [review]
Looks good

**Review by martynawitkowska — APPROVED** [review]

**Pull request merged**